GliaWidgetsTests/SecureConversations/Confirmation/SecureConversations.ConfirmationViewModelTests.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class SecureConversationsConfirmationViewModelTests: XCTestCase {
+    typealias ConfirmationViewModel = SecureConversations.ConfirmationViewModel
+    var viewModel: ConfirmationViewModel = .mock
+
+    override func setUp() {
+        viewModel = .mock
+    }
+}
+
+// Props
+extension SecureConversationsConfirmationViewModelTests {
+    func testPropsDoNotGenerateABackButton() {
+        let props = viewModel.props().confirmationViewProps.header
+
+        XCTAssertNil(props.backButton)
+    }
+
+    func testPropsGenerateCorrectTitle() {
+        let title = "Test"
+        var style = Theme().secureConversationsConfirmation
+        style.headerTitle = title
+
+        viewModel.environment = .init(confirmationStyle: style)
+
+        let props = viewModel.props()
+        XCTAssertEqual(props.confirmationViewProps.header.title, title)
+    }
+
+    func testPropsGenerateEndButtonWithAccessibilityIdentifier() {
+        let props = viewModel.props().confirmationViewProps.header.endButton
+
+        XCTAssertEqual(props.accessibilityIdentifier, "header_end_button")
+    }
+}
+
+// Delegate
+extension SecureConversationsConfirmationViewModelTests {
+    func testSendingCloseButtonEventCallsDelegate() {
+        viewModel.delegate = { event in
+            switch event {
+            case .closeTapped:
+                XCTAssertTrue(true)
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.closeTapped)
+    }
+
+    func testPressingCloseButtonCallsDelegate() {
+        viewModel.delegate = { event in
```

Comment:
I thing @yurii-glia has good point that if for some reason delegate is not called it, then these tests will not evaluate.
I would suggest to change it as [suggested above](https://github.com/salemove/ios-sdk-widgets/pull/608/files#r1183284084).